exercises/concept/windowing-system/.docs/instructions.md

```diff
@@ -0,0 +1,159 @@
+       |          ║          position.x,_                                      ║
+       |          ║          position.y  \                                     ║
+       |          ║                       \<----- size.width ----->            ║
+       |          ║                 ^      *──────────────────────┐            ║
+       |          ║                 |      │        title         │            ║
+       |          ║                 |      ├──────────────────────┤            ║
+screenSize.height ║                 |      │                      │            ║
+       |          ║            size.height │                      │            ║
+       |          ║                 |      │       contents       │            ║
+       |          ║                 |      │                      │            ║
+       |          ║                 |      │                      │            ║
+       |          ║                 v      └──────────────────────┘            ║
+       |          ║                                                            ║
+       |          ║                                                            ║
+       v          ╚════════════════════════════════════════════════════════════╝
+```
+
+## 1. Define a `size` struct for storing the dimensions of the window
+
+Define a `struct` named `size`.
+It should have two member variables `width` and `height` that store the window's current dimensions.
+The constructor function should accept initial values for these fields.
+The width is provided as the first parameter, the height as the second one.
+The default width and height should be `80` and `60`, respectively.
+
+Additionally, define a member function `void resize(int newWidth, int newHeight)` that takes a new width and height as parameters and changes the fields to reflect the new size.
+
+```cpp
+size aSize{1080, 764};
+aSize.width;
+// => 1080
+aSize.height;
+// => 764
+
+aSize.resize(1920, 1080);
+aSize.width;
+// => 1920
+aSize.height;
+// => 1080
+```
+
+## 2. Define a `position` struct to store a window position
+
+Define a struct (or class) named `position` with two member variables, `x` and `y` that store the current horizontal and vertical position, respectively, of the window's upper left corner.
+The constructor function should accept initial values for these fields.
+The value for `x` is provided as the first parameter, the value for `y` as the second one.
+The default value should be `0` for both variables.
+
+The position (0, 0) is the upper left corner of the screen with `x` values getting larger as you move right and `y` values getting larger as you move down.
+
+Also define a method `void move(int newX, int newY)` that takes new x and y parameters and changes the properties to reflect the new position.
+
+```cpp
+position point{};
+point.x;
+// => 0
+point.y;
+// => 0
+
+point.move(100, 200);
+point.x;
+// => 100
+point.y;
+// => 200
+```
+
+## 3. Define a `programWindow` class
```

Comment:
I'm not entirely sure but I think this is the only required type in the C++ track with a name in "lowerCamelCase". All other types are in "flatcase", "snake_case", or "PascalCase".
Soft vote for changing the type name to "ProgramWindow" or "program_window".